src/api/go/pxapi/client.go

```diff
@@ -87,8 +90,14 @@ func NewClient(ctx context.Context, opts ...ClientOption) (*Client, error) {
 
 func (c *Client) init(ctx context.Context) error {
 	isInternal := strings.Contains(c.cloudAddr, "cluster.local")
+	tlsDisabled := os.Getenv("PX_DISABLE_TLS") == "1"
```

Comment:
I can definitely make this a `ClientOption` -- `WithTLSVerification` or something similar. However, the motivation behind this change was to ensure that TLS verification isn't implicitly disabled (such as when `cloudAddr` is `cluster.local`). So it was intentional to make the TLS verification opt out rather than silently disabling it.

Therefore, would you be ok if downstream consumers explicitly added the `WithDisableTLSVerification` option if they needed the previous behavior?

The requirements for this came from a recent internal pentest. The goal was to document the way to opt out of TLS verification and to change the current default.

Reply:
Yeah I was thinking TLS verification by default, and then an explicit option to disable it.

Reply:
This is updated to include that `ClientOption` now. I also added unit tests for it.